ansible/roles/shared_dir/tasks/setup_host.yml

```diff
@@ -12,6 +12,20 @@
     - nfs
     - shared_files
 
+- name: Format system data disk
+  sudo: yes
+  filesystem: fstype=ext4 dev={{ datadisk_device }}1
+  when: datadisk_device is defined
+  tags:
+    - shared_files
```

Comment:
i used the `shared_files` tag because these tasks seemed to fit more with the other tasks using that tag. both get run when you deployed the shared directory on non monolith machines and this isnt specific to the nfs. if monolith machines had data disks we could do this too.